spark/v3.3/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewritePositionDeleteFiles.java

```diff
@@ -361,48 +358,43 @@ private void checkResult(
       List<DeleteFile> rewrittenDeletes,
       List<DeleteFile> newDeletes,
       int expectedGroups) {
-    Assert.assertEquals(
-        "Expected rewritten delete file count does not match",
-        rewrittenDeletes.size(),
-        result.rewrittenDeleteFilesCount());
-    Assert.assertEquals(
-        "Expected new delete file count does not match",
-        newDeletes.size(),
-        result.addedDeleteFilesCount());
-    Assert.assertEquals(
-        "Expected rewritten delete byte count does not match",
-        size(rewrittenDeletes),
-        result.rewrittenBytesCount());
-    Assert.assertEquals(
-        "Expected new delete byte count does not match",
-        size(newDeletes),
-        result.addedBytesCount());
-
-    Assert.assertEquals(
-        "Expected rewrite group count does not match",
-        expectedGroups,
-        result.rewriteResults().size());
-    Assert.assertEquals(
-        "Expected rewritten delete file count in all groups to match",
-        rewrittenDeletes.size(),
-        result.rewriteResults().stream()
-            .mapToInt(FileGroupRewriteResult::rewrittenDeleteFilesCount)
-            .sum());
-    Assert.assertEquals(
-        "Expected added delete file count in all groups to match",
-        newDeletes.size(),
-        result.rewriteResults().stream()
-            .mapToInt(FileGroupRewriteResult::addedDeleteFilesCount)
-            .sum());
-    Assert.assertEquals(
-        "Expected rewritten delete bytes in all groups to match",
-        size(rewrittenDeletes),
-        result.rewriteResults().stream()
-            .mapToLong(FileGroupRewriteResult::rewrittenBytesCount)
-            .sum());
-    Assert.assertEquals(
-        "Expected added delete bytes in all groups to match",
-        size(newDeletes),
-        result.rewriteResults().stream().mapToLong(FileGroupRewriteResult::addedBytesCount).sum());
+    assertThat(result.rewrittenDeleteFilesCount())
+        .as("Rewritten delete files")
+        .isEqualTo(rewrittenDeletes.size());
+    assertThat(result.addedDeleteFilesCount())
+        .as("Added delete files")
+        .isEqualTo(newDeletes.size());
+    assertThat(result.rewrittenBytesCount())
+        .as("Rewritten delete bytes")
+        .isEqualTo(size(rewrittenDeletes));
+    assertThat(result.addedBytesCount()).as("New Delete byte count").isEqualTo(size(newDeletes));
+
+    assertThat(result.rewriteResults().size())
```

Comment:
```suggestion
    assertThat(result.rewriteResults()).as("Rewritten group count").hasSize(expectedGroups);
```

same advantage as I mentioned above as it allows easier debugging if the assertion ever fails